sertop/sername.ml

```diff
@@ -0,0 +1,211 @@
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(************************************************************************)
+(* Coq serialization API/Plugin                                         *)
+(* Written by: Karl Palmskog                                            *)
+(************************************************************************)
+(* Status: Very Experimental                                            *)
+(************************************************************************)
+
+let fatal_exn exn info =
+  let loc = Loc.get_loc info in
+  let msg = Pp.(pr_opt_no_spc Topfmt.pr_loc loc ++ fnl ()
+                ++ CErrors.iprint (exn, info)) in
+  Format.eprintf "Error: @[%a@]@\n%!" Pp.pp_with msg;
+  exit 1
+
+let create_document ~require_lib ~in_file ~async ~async_workers ~quick ~iload_path ~debug =
+
+  let open Sertop_init in
+
+  (* coq initialization *)
+  coq_init
+    { fb_handler = (fun _ -> ())  (* XXXX *)
+    ; ml_load    = None
+    ; debug
+    };
+
+  (* document initialization *)
+
+  let stm_options = process_stm_flags
+      { enable_async  = async
+      ; deep_edits    = false
+      ; async_workers = async_workers
+      } in
+
+  (* Disable due to https://github.com/ejgallego/coq-serapi/pull/94 *)
+  let stm_options =
+    { stm_options with
+      async_proofs_tac_error_resilience = `None
+    ; async_proofs_cmd_error_resilience = false
+    } in
+
+  let stm_options =
+    if quick
+    then { stm_options with async_proofs_mode = APonLazy }
+    else stm_options
+  in
+
+  (*
+  let require_libs =
+    let prelude = ["Coq.Init.Prelude", None, Some false] in
+    match require_lib with
+    | Some l -> prelude @ [l, None, Some false]
+    | None -> prelude
+  in
+  *)
+  let require_libs = ["Coq.Init.Prelude", None, Some false] in
+
+  let ndoc = { Stm.doc_type = Stm.VoDoc in_file
+             ; require_libs
+             ; iload_path
+             ; stm_options
+             } in
+
+  (* Workaround, see
+     https://github.com/ejgallego/coq-serapi/pull/101 *)
+  if quick || async <> None
+  then Safe_typing.allow_delayed_constants := true;
+
+  match require_lib with
+  | None -> Stm.new_doc ndoc
+  | Some l ->
+     (*
+     let sdoc = Stm.new_doc ndoc in
+     let dir,from,exp = l,None,Some false in
+     let mp = Libnames.qualid_of_string dir in
+     let mfrom = Option.map Libnames.qualid_of_string from in
+     Flags.silently (Vernacentries.vernac_require mfrom exp) [mp];
+     sdoc
+     *)
+     let doc,sid = Stm.new_doc ndoc in
+     let sent = Printf.sprintf "Require %s." l in
```

Comment:
Why are you not adding this to `require_libs` above?

Reply:
You mention a problem in the commit, but with no further details given it is hard to say more.